impl/src/main/java/org/pentaho/di/trans/steps/starrockskettleconnector/starrocks/StarRocksDataType.java

```diff
@@ -0,0 +1,47 @@
+package org.pentaho.di.trans.steps.starrockskettleconnector.starrocks;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public enum StarRocksDataType {
+    TINYINT,
+    INT,
+    LARGEINT,
+    SMALLINT,
+    BOOLEAN,
+    DECIMAL,
+    DOUBLE,
+    FLOAT,
+    BIGINT,
+    VARCHAR,
+    CHAR,
+    STRING,
+    JSON,
+    DATE,
+    DATETIME,
+    UNKNOWN;
+
+    private static final Map<String, StarRocksDataType> dataTypeMap = new HashMap<>();
+    static {
+        StarRocksDataType[] starRocksDataTypes = StarRocksDataType.values();
+
+        for (StarRocksDataType starRocksDataType : starRocksDataTypes) {
+            dataTypeMap.put(starRocksDataType.name(), starRocksDataType);
+            dataTypeMap.put(starRocksDataType.name().toLowerCase(), starRocksDataType);
```

Comment:
No need to add lower-case types. It's enough to use `typeString.toUpperCase` in `fromString` method